packages/@guardian/source-foundations/src/typography/api.ts

```diff
@@ -1,110 +1,69 @@
-import { fs } from './fs';
+import { fontStyleFunction } from './font-styles';
 import type {
 	BodySizes,
-	FontScaleArgs,
-	FontScaleFunction,
 	HeadlineSizes,
 	TextSansSizes,
 	TitlepieceSizes,
+	TypographyConfiguration,
+	TypographyFunctions,
 } from './types';
 
-type TitlepieceFunctions = {
-	[key in keyof TitlepieceSizes]: FontScaleFunction;
-};
-
-const titlepieceDefaults = {
+const titlepieceDefaults: TypographyConfiguration = {
 	lineHeight: 'tight',
 	fontWeight: 'bold',
 	fontStyle: null,
 	unit: 'rem',
 };
-const titlepieceFs = fs('titlepiece');
 
-export const titlepiece: TitlepieceFunctions = {
-	small: (options?: FontScaleArgs) =>
-		titlepieceFs('small', Object.assign({}, titlepieceDefaults, options)),
-	medium: (options?: FontScaleArgs) =>
-		titlepieceFs('medium', Object.assign({}, titlepieceDefaults, options)),
-	large: (options?: FontScaleArgs) =>
-		titlepieceFs('large', Object.assign({}, titlepieceDefaults, options)),
+export const titlepiece: TypographyFunctions<TitlepieceSizes> = {
+	small: fontStyleFunction('titlepiece', 'small', titlepieceDefaults),
+	medium: fontStyleFunction('titlepiece', 'medium', titlepieceDefaults),
+	large: fontStyleFunction('titlepiece', 'large', titlepieceDefaults),
 };
 
-type HeadlineFunctions = {
-	[key in keyof HeadlineSizes]: FontScaleFunction;
-};
-const headlineDefaults = {
+const headlineDefaults: TypographyConfiguration = {
 	lineHeight: 'tight',
 	fontWeight: 'medium',
 	fontStyle: null,
 	unit: 'rem',
 };
-const headlineFs = fs('headline');
-
-export const headline: HeadlineFunctions = {
-	xxxsmall: (options?: FontScaleArgs) =>
-		headlineFs('xxxsmall', Object.assign({}, headlineDefaults, options)),
-	xxsmall: (options?: FontScaleArgs) =>
-		headlineFs('xxsmall', Object.assign({}, headlineDefaults, options)),
-	xsmall: (options?: FontScaleArgs) =>
-		headlineFs('xsmall', Object.assign({}, headlineDefaults, options)),
-	small: (options?: FontScaleArgs) =>
-		headlineFs('small', Object.assign({}, headlineDefaults, options)),
-	medium: (options?: FontScaleArgs) =>
-		headlineFs('medium', Object.assign({}, headlineDefaults, options)),
-	large: (options?: FontScaleArgs) =>
-		headlineFs('large', Object.assign({}, headlineDefaults, options)),
-	xlarge: (options?: FontScaleArgs) =>
-		headlineFs('xlarge', Object.assign({}, headlineDefaults, options)),
-};
 
-type BodyFunctions = {
-	[key in keyof BodySizes]: FontScaleFunction;
+export const headline: TypographyFunctions<HeadlineSizes> = {
+	xxxsmall: fontStyleFunction('headline', 'xxxsmall', headlineDefaults),
+	xxsmall: fontStyleFunction('headline', 'xxsmall', headlineDefaults),
+	xsmall: fontStyleFunction('headline', 'xsmall', headlineDefaults),
+	small: fontStyleFunction('headline', 'small', headlineDefaults),
+	medium: fontStyleFunction('headline', 'medium', headlineDefaults),
+	large: fontStyleFunction('headline', 'large', headlineDefaults),
+	xlarge: fontStyleFunction('headline', 'xlarge', headlineDefaults),
```

Comment:
Just wondering if there's a way to iterate over the sizes, set the key to each size and pass each size to the `fontStyleFunction` without having to write each one out individually?

Reply:
good idea! I'll have a go at implementing that tomorrow and we can compare the two approaches.
I think it'll work well, but there's a small part of me that kinda likes how declarative setting them out individually is.

My hope is that with both of them side by side it'll be an easier comparison to make :)

Reply:
@tjsilver I had a go at doing this in: [this commit: `af1ea2e` (#1525)](https://github.com/guardian/source/pull/1525/commits/af1ea2e00d1a04f7d6ecda7676556b60cda2ab7e) 

it has the added benefit of adding a strong coupling between `index.ts` and `api.ts`, meaning that if api.ts is updated, this file will reflect those changes too.

The method: `getStringStyleFunctions` takes a bit of headspace to read though, I added a doc comment which should help to explain what it does, but defining line by line does seem a bit clearer.

I'm still on the fence, what d'you think?

Reply:
I'm a bit confused, looks like your new `getStringStyleFunctions` works on the `fontStyleToStringFunction` in index.ts (which is also great), but the implementation of `fontStyleFunction` in api.ts (which I was referring to above) is left unchanged? Or is that the two options being shown together? 

Also, I see what you mean about the `getStringStyleFunctions` being a bit gnarly - could a map be used here instead of reduce, for clarity?

Reply:
oh - sorry, i completely missed that your comment was in this file and not index.ts - let's discuss on that call

Reply:
update:
we discussed the feasibility of defining the string and object style functions in the same file for simplicity, but the refactoring work would have taken longer than the time box we had decided upon.